api/metrics/v1/http.go

```diff
@@ -213,24 +213,34 @@ func NewHandler(endpoints Endpoints, upstreamCA []byte, upstreamCert *stdtls.Cer
 			}
 		}
 		r.Group(func(r chi.Router) {
+			r.Use(func(handler http.Handler) http.Handler {
```

Comment:
The instrumentation middleware is run first (line 189), then the next middleware injects the extra labels `group` and `handler` into context so that when the middleware for ratelimiter runs and when a 429 is returned back to the instrumentation handler(here -> [line101](https://github.com/observatorium/api/blob/56abe030cbbe3e50a614dacf793fa673317b937b/server/http_instrumentation.go#L101)), the ctx already has the labels for `group` and `handler` set